### PR TITLE
Move a doc out, add doctests to some things

### PR DESCRIPTION
**Pull request by kshyatt**
I changed the wording on `BoundsError` a little to make it clear that the index is the problem.

### CODE REVIEW

File: base/docs/helpdb/Base.jl [hunk 1]
```diff
@@ -986,7 +986,29 @@ finalize
...
+
+# Examples
+```jldoctest
+julia> A = ones(7);
+
+julia> A[8]
+ERROR: BoundsError: attempt to access 7-element Array{Float64,1} at index [8]
+Stacktrace:
+ [1] getindex(::Array{Float64,1}, ::Int64) at ./array.jl:586
+
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: maybe add the `[...]` here to make it clear there's truncated extra content?

Comment (kshyatt) [inline base/docs/helpdb/Base.jl]: I copy-pasted this right from the REPL, there was no extra content?

Comment (tkelman) [inline base/docs/helpdb/Base.jl]: oh the rare shallow stacktrace. repl filtered out exactly what it should've. when run as a doctest there's no complaints about the stacktrace being longer?

**Review by tkelman — APPROVED** [review]
> "What even are numbers?"

🥇

File: base/docs/helpdb/Base.jl (hunk 1 above)
Comment (kshyatt) [inline base/docs/helpdb/Base.jl]: I'll check just to make sure

Comment (kshyatt) [inline base/docs/helpdb/Base.jl]: Yep! Passed locally for me.

**Comment by kshyatt** [comment]
Is that 64 bit linux fail my fault?

**Comment by kshyatt** [comment]
Bumpity bump

**Comment by StefanKarpinski** [comment]
I've been seeing that failure all over, not related.

**Pull request merged**